.evergreen/compile-unix.sh

```diff
@@ -256,6 +256,8 @@ else
    }
 fi
 
+export MONGOC_TEST_SERVER_LOG=stdout
```

Comment:
This logs commands and replies for the mock server. I left it in since it could be helpful in diagnosing flaky test failures.